internal/services/synapse/synapse_firewall_rule_resource.go

```diff
@@ -159,12 +169,27 @@ func resourceSynapseFirewallRuleDelete(d *pluginsdk.ResourceData, meta interface
 
 	future, err := client.Delete(ctx, id.ResourceGroup, id.WorkspaceName, id.Name)
 	if err != nil {
-		return fmt.Errorf("deleting Synapse Firewall Rule %q (Workspace %q / Resource Group %q): %+v", id.Name, id.WorkspaceName, id.ResourceGroup, err)
+		return fmt.Errorf("deleting %s: %+v", id, err)
```

Comment:
this is a parsed ID, so it'll be returned as a pointer - as such this needs to be:

```suggestion
		return fmt.Errorf("deleting %s: %+v", *id, err)
```